charts/victoria-metrics-distributed/values.yaml

```diff
@@ -33,229 +38,131 @@ write:
-          spec:
-            port: "8427"
-    # vmagent here only meant to proxy write requests to each az,
-    # doesn't support customized other remote write address.
-    vmagent:
-      # -- Create VMAgent remote write proxy
-      enabled: true
-      # -- Override the name of the vmagent object
-      name: ""
-      # -- VMAgent remote write proxy annotations
-      annotations: {}
-      # -- Spec for VMAgent CRD, see [here](https://docs.victoriametrics.com/operator/api#vmagentspec)
-      spec:
-        port: "8429"
-    vmcluster:
-      # -- Create VMCluster
-      enabled: true
-      # -- Override the name of the vmcluster, by default is <zoneName>
-      name: ""
-      # -- Spec for VMCluster CRD, see [here](https://docs.victoriametrics.com/operator/api#vmclusterspec)
-      spec:
-        retentionPeriod: "14"
-        replicationFactor: 2
-        vmstorage:
-          replicaCount: 2
-          storageDataPath: "/vm-data"
-          resources: {}
-        vmselect:
-          replicaCount: 2
-          extraArgs: {}
-          port: "8481"
-          resources: {}
-        vminsert:
-          replicaCount: 2
-          extraArgs: {}
-          port: "8480"
-          resources: {}
+        spec: {}
+  # vmagent here only meant to proxy write requests to each az,
+  # doesn't support customized other remote write address.
+  vmagent:
+    # -- Create VMAgent remote write proxy
+    enabled: true
+    # -- Override the name of the vmagent object
+    name: "vmagent-{{ (.zone).name }}"
+    # -- VMAgent remote write proxy annotations
+    annotations: {}
+    # -- Spec for VMAgent CRD, see [here](https://docs.victoriametrics.com/operator/api#vmagentspec)
+    spec: {}
+  vmcluster:
+    # -- Create VMCluster
+    enabled: true
+    # -- Override the name of the vmcluster, by default is <zoneName>
+    name: "vmcluster-{{ (.zone).name }}"
+    # -- Spec for VMCluster CRD, see [here](https://docs.victoriametrics.com/operator/api#vmclusterspec)
+    spec:
+      retentionPeriod: "14"
+      replicationFactor: 2
+      vmstorage:
+        replicaCount: 2
+        storageDataPath: "/vm-data"
+        resources: {}
+      vmselect:
+        replicaCount: 2
+        extraArgs: {}
+        resources: {}
+      vminsert:
+        replicaCount: 2
+        extraArgs: {}
+        resources: {}
+
+# -- Config per availability zone components, including vmagent, vmcluster, vmauth etc
+# Each zone uses config from `zoneTpl` by default, override it for each availability zone if it's needed
+availabilityZones:
+  -
+    # -- Availability zone name
+    name: zone-eu-1
+  - 
+    # -- Availability zone name
+    name: zone-us-1
 
 
 # -- Set up an extra vmagent to scrape all the scrape objects by default,
-# and write data to above vmauth-global-ingest endpoint.
-extraVMAgent:
-  enabled: true
-  name: test-vmagent
-  spec:
-    selectAllByDefault: true
+# and write data to above write-global endpoint.
+extra:
+  vmagent:
+    enabled: false
```

Comment:
I’d like to keep it enabled by default to write data to the write-global endpoint, allowing users to see data the first time they try this chart, and they can always disable it later.

Reply:
updated